src/lavinmq/mfile.cr

```diff
@@ -90,9 +90,12 @@ class MFile < IO
     addr
   end
 
-  def delete : self
-    code = LibC.unlink(@path.check_no_null_byte)
-    raise File::Error.from_errno("Error deleting file", file: @path) if code < 0
+  def delete(*, raise_on_missing = true) : self
+    if raise_on_missing
+      File.delete(@path)
```

Comment:
`File.delete` (public API) is using `Crystal::System::File.delete` (non-public API) which will use platform dependent calls. On unix it will be `LibC.unlink`. Now `MFile` doesn't work on anything else than *nix, but I think it's nice to use public APIs.

Reply:
https://github.com/crystal-lang/crystal/blob/master/src/file.cr#L427
and
https://github.com/crystal-lang/crystal/blob/master/src/crystal/system/unix/file.cr#L131-L140

Reply:
aha ok, thanks for the explanation